frontend/src/AppState/AppState.tsx

```diff
@@ -1,78 +1,153 @@
-import {SiteInfo, SiteWithUserInfo} from '../Types/SiteInfo';
+import {SiteWithUserInfo} from '../Types/SiteInfo';
+import {observable, makeObservable, autorun, action} from 'mobx';
+import APICache from '../API/APICache';
 
-export enum AppState {
+export enum AppLoadingState {
     loading,
     unauthorized,
     authorized
 }
 
-export type UserStats = {
+export type UserStatus = {
     watch: {
         posts: number;
         comments: number;
     };
     notifications: number;
+    subscriptions: SiteWithUserInfo[];
 };
 
 type AppStateContextState = {
-    userInfo?: UserInfo;
     appState: AppState;
-    api: APIHelper;
-    site?: SiteWithUserInfo;
-    userStats: UserStats;
 };
 
-export type AppStateSetters = {
-    setSite: Dispatch<SetStateAction<SiteWithUserInfo | undefined>>;
-    setUserInfo: (user: UserInfo | undefined) => void;
-    setAppState: (state: AppState) => void;
-    setUserStats: Dispatch<SetStateAction<UserStats>>;
-};
+export class AppState {
+    @observable
+    appLoadingState = AppLoadingState.loading;
 
-const AppStateContext = createContext<AppStateContextState>({} as AppStateContextState);
+    @observable.struct
+    userInfo: UserInfo | undefined = undefined; // undefined means not authorized
 
-const apiBase = new APIBase();
+    @observable.deep
+    userStatus: UserStatus = { notifications: 0, watch: { posts: 0, comments: 0 }, subscriptions: [] };
 
-export const AppStateProvider = (props: {children: ReactNode}) => {
-    const [userInfo, setUserInfo] = useState<UserInfo | undefined>(undefined);
-    const [appState, setAppState] = useState<AppState>(AppState.loading);
-    const [userStats, setUserStats] = useState<UserStats>({ notifications: 0, watch: { posts: 0, comments: 0 } });
-    const [site, setSite] = useState<SiteInfo>();
-    const api = useMemo(() => new APIHelper(apiBase, {setUserInfo, setAppState, setSite, setUserStats}), []);
-    api.updateSetters({setUserInfo, setAppState, setSite, setUserStats});
+    @observable.struct
+    siteInfo: SiteWithUserInfo | undefined = undefined;
 
-    useEffect(() => {
-        api.init().then();
-    }, [api]);
+    @observable
+    site = 'main';
 
-    useEffect(() => {
-        const link = document.querySelector('link[rel~="icon"]') as HTMLLinkElement;
-        if (!link) {
-            return;
+    readonly api: APIHelper;
+    readonly cache: APICache;
+
+    constructor() {
+        makeObservable(this);
+
+        const apiBase = new APIBase();
+        this.cache = new APICache();
+        this.api = new APIHelper(apiBase, this);
+        this.api.init().then().catch();
+    }
+
+    @action
+    setAppLoadingState(value: AppLoadingState) {
+        this.appLoadingState = value;
+    }
+
+    @action
+    setUserInfo(value: UserInfo | undefined) {
+        this.userInfo = value;
+    }
+
+    @action
+    setUserStatus(value: UserStatus) {
+        if (value.notifications !== this.userStatus.notifications) {
```

Comment:
Why not use  `observable.struct` for `userStatus` instead of manual deep equality checks?